### PR TITLE
Deprecate Tuple as a value of on keyword argument in join

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/1915

### CODE REVIEW

**Comment by bkamins** [comment]
CI fails due to coverage only

File: src/abstractdataframe/join.jl [hunk 1]
```diff
@@ -298,6 +297,10 @@ function Base.join(df1::AbstractDataFrame,
                    validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false))
     _check_consistency(df1)
     _check_consistency(df2)
+    if on isa NTuple{2,Symbol} || on isa AbstractVector{NTuple{2,Symbol}}
+        Base.depwarn("Using a `Tuple{Symbol, Symbol} or a vector of such tuples " *
+                     "as a value of `on` keyword argument is deprecated.", :join)
```
Comment (nalimilan) [inline src/abstractdataframe/join.jl]:
```suggestion
                     "as a value of `on` keyword argument is deprecated: use pairs instead.", :join)
```

File: src/abstractdataframe/join.jl [hunk 2]
```diff
@@ -298,6 +297,10 @@ function Base.join(df1::AbstractDataFrame,
                    validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false))
     _check_consistency(df1)
     _check_consistency(df2)
+    if on isa NTuple{2,Symbol} || on isa AbstractVector{NTuple{2,Symbol}}
+        Base.depwarn("Using a `Tuple{Symbol, Symbol} or a vector of such tuples " *
```
Comment (nalimilan) [inline src/abstractdataframe/join.jl]:
```suggestion
        Base.depwarn("Using a `Tuple{Symbol, Symbol}` or a vector of such tuples " *
```

**Comment by bkamins** [comment]
@nalimilan - thank you!

**Pull request merged**